etcd: fix JWT renewal with user/pass authentication

awly opened the pull request:
Two changes here:

1. update go.etcd.io/etcd to their latest module-compliant versions
2. use a separate connection for peer version checking
    the implementation of `b.client.Status` (in the etcd library) internally causes the in-memory JWT store to be overwritten.
    this ends up breaking the JWT refresh logic (one store has the exired token but a different one gets the refreshed one).
    I will send an upstream fix too

\# 2 works without # 1.
Since this PR is against `master`, I decided to keep # 1 anyway, for 7.0.
I will cherry-pick only # 2 to 6.2, without dependency changes.

FYI @fspmarshall because you had some memory issues with the etcd client; check if this updated version of the library fixes them

Fixes https://github.com/gravitational/teleport/issues/6881

Joerger commented:
@klizhentas has shown concern about moving to a beta version of etcd for a production release #6183. However it doesn't seem clear when etcd plans on releasing 3.5 and this has pushed back a few other fixes as well. @awly Do you have a sense of how stable the current beta version is or how soon it'll be released officially?

r0mant commented on lib/backend/etcdbk/etcd.go:
```diff
@@ -292,6 +298,12 @@ func (b *EtcdBackend) CloseWatchers() {
 func (b *EtcdBackend) reconnect(ctx context.Context) error {
+	if b.client != nil {
+		if err := b.client.Close(); err != nil {
+			b.Entry.WithError(err).Warningf("Failed closing existing etcd client on reconnect.")
```
Nit: 
```suggestion
			b.Entry.WithError(err).Warning("Failed closing existing etcd client on reconnect.")
```

r0mant reviewed (commented):
Change lgtm but I agree I would be cautious vendoring an alpha dependency. Is it required for this change?

Commit:
etcd: use a separate connection to check peer versions

 There is a data race in etcd that breaks the internal state in etcd client
implementation for some server setups (user/pass authentication with
JWTs).

awly commented:
You're right, let's wait for etcd 3.5 to be fully released before updating.
Reverted all dependency changes. PTAL

Commit:
Merge branch 'master' into andrew/update-etcd
